### DISCUSSION TITLE
Why does renovate create incompatibility in `@types/ws` packages?

### DISCUSSION THREAD

**Discussion by oliversalzburg** [renovatebot/renovate#10523] [Request Help] [answered]
Consider this small reproduction: https://github.com/oliversalzburg/renovate-ws-types What I'm observing in my full repository should be similar to the state in the reproduction.

I have a package depending on `@types/ws`, which was previously resolved to 7.4.4. My own project depends on `@types/ws` as well.  
When Renovate now attempts to update `@types/ws`, it will only upgrade my own dependency version to 7.4.5 and create a second entry in the lockfile to maintain the nested dependency at version 7.4.4. This makes my code incompatible with the dependency code, as the types are incompatible.

The descriptor for `@types/ws` in the dependency (`^7.0.0`) would allow for it to be upgraded to 7.4.5 as well.

Why does this happen and how can I control this behavior?

**Comment by rarkins** [comment, marked as answer]
The lock file is generated by `yarn` itself, not Renovate. You may want to look into the `yarnDedupeFewer` option in [`postUpdateOptions`](https://docs.renovatebot.com/configuration-options/#postupdateoptions).

**Reply by oliversalzburg** [reply]
That's interesting. Thanks.

I tried that option, but it had no effect on my repo. However, when I check out the `renovate/ws-7.x` branch locally and run `yarn dedupe` the entries for `@types/ws` are deduplicated as I would expect.

I noticed the documentation only mentions `yarn dedupe` in relation to `yarnDedupeHighest` and refers only to `yarn-deduplicate` for `yarnDedupeFewer`. But `yarn-deduplicate` doesn't support yarn v2 lockfiles apparently.

Am I missing something?

**Reply by oliversalzburg** [reply]
Switching to `yarnDedupeHighest` did dedupe the lockfile. Sadly this now leads to new issues, but I can see that this is technically the only working approach right now. I'll probably upgrade these types manually for now. Thanks for your support.
